windows-agent/internal/proservices/proservices_test.go

```diff
@@ -63,15 +64,25 @@ func TestNew(t *testing.T) {
 			}
 
 			s, err := proservices.New(ctx, publicDir, privateDir, proservices.WithRegistry(reg))
-			if err == nil {
-				defer s.Stop(ctx)
-			}
-
 			if tc.wantErr {
 				require.Error(t, err, "New should return an error")
 				return
 			}
 			require.NoError(t, err, "New should return no error")
+			defer s.Stop(ctx)
+
+			// defer is needed here because the registry mock will defer notifying the registry watcher after the following writes.
+			defer func() {
+				got, err := os.ReadFile(filepath.Join(publicDir, ".cloud-init", "agent.yaml"))
+				require.NoError(t, err, "Setup: could not read agent.yaml file post test completion")
+				want := testutils.LoadWithUpdateFromGolden(t, string(got))
+				require.Equal(t, want, string(got), "agent.yaml file should be the same as the golden file")
+			}()
+			err = reg.WriteValue(k, "LandscapeConfig", "[client]\nuser=JohnDoe", true)
```

Comment:
Maybe you should add a comment that those will never write anything to the config files, but just to trigger the callbacks as the tests cases disallow writing content to the agent.yaml file? (So that we don’t question ourself in the future on why the Goldens are empty). Of course, if we update the test case for one valid use case later, then, this will need to be reworked.

Reply:
I'll need to park another task just to refactor this part to find a reliable way (not time-based) of asserting that cloud-init would have time to be notified and write the agent.yaml.

So for now let's go with the empty goldens and add the comment.